gradoop-flink/src/main/java/org/gradoop/flink/io/impl/csv/CSVDataSource.java

```diff
@@ -50,11 +53,29 @@ public CSVDataSource(String csvPath, GradoopFlinkConfig config) {
     super(csvPath, config);
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * graph heads will be disposed.
+   */
   @Override
   public LogicalGraph getLogicalGraph() {
+    return getGraphCollection().reduce(new ReduceCombination());
```

Comment:
Using a `ReduceCombination` will create a new graph head and therefore discard graph head properties.

Reply:
The reduce combination used when reading a single logical graph is the way most data sources work.
I opened Issue #974 to discuss improvements. Regarding this pull request, I would keep it this way and change this behavior for all the affected data sources.

Reply:
Okay, I forgot about that issue. So yes, leave it like this.